directory/directory.go

```diff
@@ -114,3 +113,74 @@ func (d *Directory) DeleteTodo(ctx context.Context, id string) error {
 
 	return nil
 }
+
+func (d *Directory) resolveIdentity(ctx context.Context, identity string) (*dsc.Object, error) {
+	if d.isLegacy {
+		return d.resolveIdentityLegacy(ctx, identity)
+	}
+
+	relResp, err := d.Reader.GetRelation(ctx, &dsr.GetRelationRequest{
+		SubjectType: IdentityObjectType,
+		SubjectId:   identity,
+		Relation:    IdentifierRelationType,
+		ObjectType:  UserObjectType,
+		WithObjects: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	user, ok := relResp.Objects[fmt.Sprintf("%s:%s", "user", relResp.Result.ObjectId)]
```

Comment:
Don't use `"user"` as a literal. It should be either `UserObjectType` or `relResp.Result.ObjectType`.